presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemorySplitManager.java

```diff
@@ -43,17 +43,18 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
     {
         MemoryTableLayoutHandle layout = (MemoryTableLayoutHandle) layoutHandle;
 
-        List<HostAddress> hosts = layout.getTable().getHosts();
+        Map<HostAddress, Long> rowsOnHosts = layout.getHosts();
```

Comment:
This could also be wrappered for better visibility - something like - class InMemoryFragment { HostAddress, long }